linodecli/plugins/k8s-alpha.py

```diff
@@ -281,6 +298,37 @@ def dep_installed(command):
     except: 
         return False
 
+def terraform_version_supported(min_version, max_version):
+    # The Terraform version is of the format "Terraform v0.0.0"
+    version = check_output(['terraform','version']).split()[1].replace('v','')
+    return terraform_minimum_check(version, min_version) and terraform_maximum_check(version, max_version)
+
+def terraform_minimum_check(version, min_version):
+    version = [int(i) for i in version.split('.')]
+    minimum = [int(i) for i in min_version.split('.')]
+    # Check that version is equal to or greater than minimum
+    for i in range(len(version)):
+        if version[i] < minimum[i]:
+            return False
+        elif version[i] > minimum[i]:
+            return True
+        else:
+            pass
+    return True
+
+def terraform_maximum_check(version, max_version):
+    version = [int(i) for i in version.split('.')]
+    maximum = [int(i) for i in max_version.split('.')]
+    # Check that version is less than maximum
+    for i in range(len(version)):
+        if version[i] < maximum[i]:
+            return True
+        elif version[i] > maximum[i]:
+            return False
+        else:
+            pass
+    return False
```

Comment:
You might consider using `LooseVersion` (or `StrictVersion` if you're confident the version numbers will always be in a sane format) to make these checks simpler:

```python
def terraform_version_supported(min_version, max_version):
  version = check_output(['terraform','version']).split()[1].replace('v','')
  return LooseVersion(min_version) < LooseVersion(version) < LooseVersion(max_version)
```

(That example is probably oversimplified, I didn't actually test it)

The CLI already does something similar when checking for updates [here](https://github.com/linode/linode-cli/blob/cd291f6ae3a3c2c0bbbd738a6ab315465a885efd/linodecli/cli.py#L429)

Reply:
Good suggestion! I will definitely make use of the `distutil` library.

Also, I looked a bit more into Terraform versions, and it seems they do stick with semantic versioning. https://github.com/hashicorp/terraform/blob/master/version/version.go#L13-L19

While the release versions are just integers, people can download alpha, beta, rc, etc releases on their own. Here is an example of one that is `Terraform v0.12.0-rc1`. https://releases.hashicorp.com/terraform/0.12.0-rc1/

I'll update the code accordingly.